src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonPropertyInfoOfT.cs

```diff
@@ -372,7 +371,7 @@ internal override bool ReadJsonAsObject(ref ReadStack state, ref Utf8JsonReader
             bool isNullToken = reader.TokenType == JsonTokenType.Null;
             if (isNullToken && !TypedEffectiveConverter.HandleNullOnRead && !state.IsContinuation)
             {
-                if (!PropertyTypeCanBeNull)
```

Comment:
should we remove this field overall or we still need it for non-generic JsonTypeInfo?

Reply:
non-generic still needs it, but none of the uses are part of the hot path.